Nil pointer reference when using network-operator provider

/kind bug

**What steps did you take and what happened:**
Using the latest version `v1.2.0`, seeing this error in the logs
```log
0604 01:08:59.063021       1 reflector.go:219] Starting reflector *v1.Secret (9m24.234605348s) from pkg/mod/k8s.io/client-go@v0.23.0/tools/cache/reflector.go:167
I0604 01:08:59.063085       1 reflector.go:255] Listing and watching *v1.Secret from pkg/mod/k8s.io/client-go@v0.23.0/tools/cache/reflector.go:167
I0604 01:08:59.148214       1 shared_informer.go:270] caches populated
panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x18 pc=0x7c2d36]

goroutine 872 [running]:
github.com/go-logr/logr.Logger.Info({{0x0, 0x0}, 0x21b6bfc}, {0x21f48b5, 0x3b}, {0xc000216d40, 0x1, 0x1})
	/go/pkg/mod/github.com/go-logr/logr@v1.2.0/logr.go:245 +0x56
sigs.k8s.io/cluster-api-provider-vsphere/pkg/services/network.(*netopNetworkProvider).getDefaultClusterNetwork(0x0, 0xc000b04400)
	/workspace/pkg/services/network/netop_provider.go:58 +0xa5
sigs.k8s.io/cluster-api-provider-vsphere/pkg/services/network.(*netopNetworkProvider).getClusterNetwork(...)
```


**What did you expect to happen:**
CAPV controller show not panic or fail

**Anything else you would like to add:**
n/a

**Environment:**

- Cluster-api-provider-vsphere version: `v1.2.0`

--- a/controllers/vspheremachine_controller.go
+++ b/controllers/vspheremachine_controller.go
@@ -181,7 +181,7 @@ func (r machineReconciler) Reconcile(ctx goctx.Context, req ctrl.Request) (_ ctr
 		return reconcile.Result{}, err
 	}
 	if machine == nil {
-		r.Logger.V(2).Info("waiting on Machine controller to set OwnerRef on infra machine")
+		logger.V(2).Info("waiting on Machine controller to set OwnerRef on infra machine")
 		return reconcile.Result{}, nil
 	}
 
@@ -200,7 +200,7 @@ func (r machineReconciler) Reconcile(ctx goctx.Context, req ctrl.Request) (_ ctr
 		ControllerContext: r.ControllerContext,
 		Cluster:           cluster,
 		Machine:           machine,
-		Logger:            r.Logger.WithName(req.Namespace).WithName(req.Name),
+		Logger:            logger,
 		PatchHelper:       patchHelper,
 	})
 	// always patch the VSphereMachine object
@@ -231,9 +231,9 @@ func (r machineReconciler) Reconcile(ctx goctx.Context, req ctrl.Request) (_ ctr
 	}
 
 	// Fetch the VSphereCluster and update the machine context
-	machineContext, err = r.VMService.FetchVSphereCluster(r.Client, cluster, r.ControllerContext, machineContext)
+	machineContext, err = r.VMService.FetchVSphereCluster(r.Client, cluster, machineContext)
 	if err != nil {
-		r.Logger.Info("unable to retrieve VSphereCluster", "error", err)
+		logger.Info("unable to retrieve VSphereCluster", "error", err)
 		return reconcile.Result{}, nil
 	}
 
@@ -355,7 +355,7 @@ func (r *machineReconciler) setVMModifiers(c context.MachineContext) error {
 		// No need to check the type. We know this will be a VirtualMachine
 		vm, _ := obj.(*vmoprv1.VirtualMachine)
 		ctx.Logger.V(3).Info("Applying network config to VM", "vm-name", vm.Name)
-		err := r.networkProvider.ConfigureVirtualMachine(ctx.ClusterContext, vm)
+		err := r.networkProvider.ConfigureVirtualMachine(ctx.GetClusterContext(), vm)
 		if err != nil {
 			return nil, errors.Errorf("failed to configure machine network: %+v", err)
 		}

--- a/pkg/context/vmware/machine_context.go
+++ b/pkg/context/vmware/machine_context.go
@@ -33,7 +33,6 @@ type VMModifier func(runtime.Object) (runtime.Object, error)
 // SupervisorMachineContext is a Go context used with a VSphereMachine.
 type SupervisorMachineContext struct {
 	*context.BaseMachineContext
-	ClusterContext *ClusterContext
 	VSphereCluster *vmwarev1.VSphereCluster
 	VSphereMachine *vmwarev1.VSphereMachine
 	VMModifiers    []VMModifier
@@ -55,6 +54,15 @@ func (c *SupervisorMachineContext) GetVSphereMachine() context.VSphereMachine {
 
 func (c *SupervisorMachineContext) GetObjectMeta() v1.ObjectMeta {
 	return c.VSphereMachine.ObjectMeta
+}
+
+func (c *SupervisorMachineContext) GetClusterContext() *ClusterContext {
+	return &ClusterContext{
+		ControllerContext: c.ControllerContext,
+		Cluster:           c.Cluster,
+		VSphereCluster:    c.VSphereCluster,
+		Logger:            c.GetLogger(),
+	}
 }
 
 func (c *SupervisorMachineContext) SetBaseMachineContext(base *context.BaseMachineContext) {

--- a/pkg/services/interfaces.go
+++ b/pkg/services/interfaces.go
@@ -31,7 +31,7 @@ import (
 // VSphereMachineService is used for vsphere VM lifecycle and syncing with VSphereMachine types.
 type VSphereMachineService interface {
 	FetchVSphereMachine(client client.Client, name types.NamespacedName) (context.MachineContext, error)
-	FetchVSphereCluster(client client.Client, cluster *clusterv1.Cluster, controllerContext *context.ControllerContext, machineContext context.MachineContext) (context.MachineContext, error)
+	FetchVSphereCluster(client client.Client, cluster *clusterv1.Cluster, machineContext context.MachineContext) (context.MachineContext, error)
 	ReconcileDelete(ctx context.MachineContext) error
 	SyncFailureReason(ctx context.MachineContext) (bool, error)
 	ReconcileNormal(ctx context.MachineContext) (bool, error)

--- a/pkg/services/vimmachine.go
+++ b/pkg/services/vimmachine.go
@@ -49,7 +49,7 @@ func (v *VimMachineService) FetchVSphereMachine(c client.Client, name types.Name
 	return &context.VIMMachineContext{VSphereMachine: vsphereMachine}, err
 }
 
-func (v *VimMachineService) FetchVSphereCluster(c client.Client, cluster *clusterv1.Cluster, controllerContext *context.ControllerContext, machineContext context.MachineContext) (context.MachineContext, error) {
+func (v *VimMachineService) FetchVSphereCluster(c client.Client, cluster *clusterv1.Cluster, machineContext context.MachineContext) (context.MachineContext, error) {
 	ctx, ok := machineContext.(*context.VIMMachineContext)
 	if !ok {
 		return nil, errors.New("received unexpected VIMMachineContext type")

--- a/pkg/services/vmoperator/vmopmachine.go
+++ b/pkg/services/vmoperator/vmopmachine.go
@@ -51,31 +51,23 @@ type VmopMachineService struct {
 func (v *VmopMachineService) FetchVSphereMachine(client client.Client, name types.NamespacedName) (context.MachineContext, error) {
 	vsphereMachine := &vmwarev1.VSphereMachine{}
 	err := client.Get(goctx.Background(), name, vsphereMachine)
-
 	return &vmware.SupervisorMachineContext{VSphereMachine: vsphereMachine}, err
 }
 
-func (v *VmopMachineService) FetchVSphereCluster(c client.Client, cluster *clusterv1.Cluster, controllerContext *context.ControllerContext, machineContext context.MachineContext) (context.MachineContext, error) {
+func (v *VmopMachineService) FetchVSphereCluster(c client.Client, cluster *clusterv1.Cluster, machineContext context.MachineContext) (context.MachineContext, error) {
 	ctx, ok := machineContext.(*vmware.SupervisorMachineContext)
 	if !ok {
 		return nil, errors.New("received unexpected SupervisorMachineContext type")
 	}
+
 	vsphereCluster := &vmwarev1.VSphereCluster{}
-	vsphereClusterName := client.ObjectKey{
+	key := client.ObjectKey{
 		Namespace: machineContext.GetObjectMeta().Namespace,
 		Name:      cluster.Spec.InfrastructureRef.Name,
 	}
-
-	err := c.Get(goctx.Background(), vsphereClusterName, vsphereCluster)
-
-	clusterContext := &vmware.ClusterContext{
-		ControllerContext: controllerContext,
-		Cluster:           cluster,
-		VSphereCluster:    vsphereCluster,
-	}
+	err := c.Get(goctx.Background(), key, vsphereCluster)
 
 	ctx.VSphereCluster = vsphereCluster
-	ctx.ClusterContext = clusterContext
 	return ctx, err
 }
 
@@ -564,7 +556,7 @@ func getVMLabels(ctx *vmware.SupervisorMachineContext, vmLabels map[string]strin
 
 	// Get the labels for the VM that differ based on the cluster role of
 	// the Kubernetes node hosted on this VM.
-	clusterRoleLabels := clusterRoleVMLabels(ctx.ClusterContext, infrautilv1.IsControlPlaneMachine(ctx.Machine))
+	clusterRoleLabels := clusterRoleVMLabels(ctx.GetClusterContext(), infrautilv1.IsControlPlaneMachine(ctx.Machine))
 	for k, v := range clusterRoleLabels {
 		vmLabels[k] = v
 	}

--- a/pkg/util/testutil.go
+++ b/pkg/util/testutil.go
@@ -173,7 +173,6 @@ func CreateMachineContext(clusterContext *vmware.ClusterContext, machine *cluste
 			Machine: machine,
 			Cluster: clusterContext.Cluster,
 		},
-		ClusterContext: clusterContext,
 		VSphereCluster: clusterContext.VSphereCluster,
 		VSphereMachine: vsphereMachine,
 	}

